pyriemann/datasets/sampling.py

```diff
@@ -0,0 +1,338 @@
+from functools import partial
+import numpy as np
+from sklearn.utils import check_random_state
+from pyriemann.utils.base import sqrtm
+
+
+def _pdf_r(r, sigma):
+    """pdf for the log of eigenvalues of a SPD matrix
+
+    Probability density function for the logarithm of the eigenvalues of a SPD
+    matrix samples from the Riemannian Gaussian distribution. See Said et al.
+    "Riemannian Gaussian distributions on the space of symmetric positive
+    definite matrices" (2017) for the mathematical details.
+
+    Parameters
+    ----------
+    r : ndarray shape (n_dim,)
+        Vector defines in R^n_dim.
+    sigma : float
+        Dispersion of the Riemannian Gaussian distribution.
+
+    Returns
+    -------
+    p : float
+        Probability density function applied to data point r
+    """
+
+    n_dim = len(r)
+    partial_1 = -np.sum(r**2)/sigma**2
```

Comment:
`sigma` should be checked as positive. But is it really useful, since it is squared here?

Reply:
Not really useful for the calculations, but conceptually relevant.